### PR TITLE
server: Simplify `CubebDeviceCollectionManager`.

### PR DESCRIPTION
**Pull request by kinetiknz**
Rather than including a `devtype` on the `DeviceCollectionChangeCallbacks` and using it inside `CubebDeviceCollectionMananger` to maintain a unique list of callbacks, use a pair of `(DeviceCollectionChangeCallbacks, DeviceType)` for each register/unregister call, making the overall implementation of `CubebDeviceCollectionManager` much simpler.

Now that CubebDeviceCollectionManager no longer needs to mutate DeviceCollectionChangeCallbacks, it's possible to drop the `RefCell` wrapping it.

### CODE REVIEW

**Review by ChunMinChang — APPROVED** [review]
LGTM

File: server/src/server.rs [hunk 1]
```diff
@@ -42,63 +42,42 @@ impl CubebDeviceCollectionManager {
...
-            servers.retain(|s| !Rc::ptr_eq(s, server));
-        }
+        servers.retain(|(s, d)| !Rc::ptr_eq(s, server) || d != &devtype);
         if servers.is_empty() {
             self.internal_register(context, false)?;
         }
         Ok(())
     }
 
     fn unregister_server(
```
Comment (ChunMinChang) [inline server/src/server.rs:72]: Somehow my comment here is lost.

I think we can entirely remove this function now. We can use `self.unregister(context, server, cubeb::DeviceType::all())` directly in `CubebServer::drop`

Comment (kinetiknz) [reply server/src/server.rs:72]: Good idea, I'll push this in a follow up.

Comment (kinetiknz) [reply server/src/server.rs:72]: https://github.com/mozilla/audioipc/commit/44ade28aa69e90103318f2ad501a6962c9413985

**Pull request merged**